\section{Introduction}

Phylodynamic modeling provides a powerful framework for linking evolutionary processes with population dynamics by leveraging genetic data to study how lineages diversify, spread, and change over time. This integrative approach has been extensively applied across both epidemiology and macroevolution. In epidemiology, phylodynamics enables the estimation of key epidemiological parameters, such as the basic reproduction number, \( R_t \), which quantifies the average number of secondary infections caused by a single infected individual in a fully susceptible population. These methods have been instrumental in understanding the spread of pathogens including SARS-CoV-2, Ebola, Zika, and HIV~\cite{stadler2014insights, vasylyeva2019tracing, giovanetti2020genomic, seemann2020tracking}. In macroevolution, phylodynamic approaches allow inference of speciation and extinction rates, offering insights into the diversification dynamics of clades~\cite{morlon2014phylogenetic, stadler2013can, nee1994reconstructed}.

Beyond simply estimating diversification rates, these methods can be used to test hypotheses regarding temporal variation in evolutionary rates, the impact of ecological factors on lineage dynamics, and trait-dependent diversification. For example, birth-death skyline (BDSKY) models permit the reconstruction of time-varying evolutionary processes in epidemiological systems~\cite{stadler2013birth}, while Bayesian analysis of macroevolutionary mixtures (BAMM) allows the detection and quantification of heterogeneity in speciation and extinction rates across phylogenetic trees, accounting for both temporal and lineage-specific variation~\cite{rabosky2013rates, rabosky2014analysis}. Multi-type birth-death (MTBD) models have further extended these approaches by enabling the estimation of distinct evolutionary dynamics for different compartments or subpopulations within a single system~\cite{kuhnert2016phylodynamics, stadler2013uncovering}. In macroevolutionary research, state-speciation-extinction (SSE) models have become widely used for exploring how trait variation influences diversification rates, providing a mechanistic link between organismal traits and lineage dynamics~\cite{maddison2007estimating, beaulieu2016detecting}. Additionally, generalized linear models (GLMs) have been employed to integrate external non-genetic data into phylodynamic analyses, and are frequently used in epidemiology to model the effect of covariates such as host mobility or environmental factors on parameters like migration rates between subpopulations~\cite{lemey2014unifying, muller2019inferring, valenzuela2021comprehensive}.

Despite these important advances, current approaches remain constrained in several ways. BAMM, BDSKY, and MTBD models are not inherently hypothesis-driven and cannot directly incorporate external covariates. SSE models are restricted to analyses based solely on trait data, while GLMs are limited by their functional form and often fail to capture complex interactions between variables. To address these challenges, we introduce a new framework based on Bayesian neural networks (BNNs). Neural networks, a class of graphical models, have gained increasing attention due to their capacity as universal approximators, capable of modeling arbitrarily complex functions when provided with sufficient parameters~\cite{haykin1994neural}. Typically, they are employed in supervised learning, where mappings between inputs and outputs are inferred from training data. However, in phylodynamic contexts, generating comprehensive training datasets is impractical due to the vast diversity of possible evolutionary scenarios. While recent studies have explored BNNs in the analysis of fossil and non-phylogenetic data~\cite{hauffe2024trait}, we propose extending their application to phylogenetic and molecular inference, offering a flexible and powerful alternative to existing approaches.

Through extensive simulations, we demonstrate that our framework reliably recovers the true relationships between non-genetic covariates and phylodynamic parameters, avoids overfitting despite its large parameter space, and matches the performance of GLMs in linear settings while significantly outperforming them under nonlinear dynamics. Using tools from explainable AI, we further show that the model can identify which predictors meaningfully contribute to epidemiological or macroevolutionary dynamics, providing both accurate inference and interpretability. Finally, we validate the method on empirical datasets, estimating migration rates underlying COVID-19 early spread in Europe from viral sequence alignments and mobility data and inferring speciation and extinction dynamics across living and extinct platyrrhine species. The results of our analyses highlight the flixibility and robustness of our approach, unlocking new avenues to model complex real-world epidemiological and macroevolutionary dynamics.
